File: frontend-app/src/app/types/workout/workoutDTO.ts
export interface ExerciseDTO {
  id: string;
  name: string;
  sets: number;
  reps: number;
  weight?: number;
  restTimeSec?: number;
}

export interface WorkoutDTO {
  id: string;
  title: string;
  scheduledAt: string;
  completedAt?: string | null;
  exercises: ExerciseDTO[];
  muscleGroups?: string[];
  mainFocus?: string;
}

File: backend-app/apiModule/ApiModule/WebApplication1/Api/Contracts/WorkoutDto.cs
namespace ApiModule.Api.Contracts;

public sealed class WorkoutDto
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string ScheduledAt { get; set; } = string.Empty;
    public string? CompletedAt { get; set; }

    public string[] MuscleGroups { get; set; } = Array.Empty<string>();
    public string? MainFocus { get; set; }
    public string? PerceivedLoad { get; set; }

    public List<ExerciseDto> Exercises { get; set; } = new();
}
